### PR TITLE
pull in openshift/library updates, including jenkins agent streams

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

**Comment by gabemontero** [comment]
image-eco was terraform / aws network hiccups on startup

**Comment by gabemontero** [comment]
Hmm ... the new agent imagestrem / image ref stuff needs some investigating / unit testing ... isn't turn key as is

First, the envs we expected are not there on startup:
```
time="2019-01-30T21:01:59Z" level=warning msg="The environment variable IMAGE_AGENT_MAVEN was not set and we cannot update the jenkins-agent-maven:v4.0 image references"
time="2019-01-30T21:01:59Z" level=warning msg="The environment variable IMAGE_AGENT_NODEJS was not set and we cannot update the jenkins-agent-nodejs:v4.0 image references"
time="2019-01-30T21:01:59Z" level=warning msg="The environment variable IMAGE_JENKINS was not set and we cannot update the jenkins:2 image references"
```

Also only seeing `jenkins-agent-` in the in progress reason field.

**Comment by gabemontero** [comment]
I'm seeing the env's set at least locally now ... perhaps a question of moving parts under the api.ci and various release plumbing

Will start testing locally

Have not seen e2e-aws-operator test have its cluster successfully started up yet ... will retest in a bit

**Comment by gabemontero** [comment]
found the current issue ... the import in progress reason field maintenance could not deal with `jenkins-agent-nodejs` vs. `nodejs`

coding fix now

**Comment by gabemontero** [comment]
commit to fix https://github.com/openshift/cluster-samples-operator/pull/88#issuecomment-459444925 pushed 

@bparees ptal

File: pkg/stub/jenkins.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ func tagInPayload(tag, env string, stream *imagev1.ImageStream) *imagev1.ImageSt
 	}
 	for _, tagSpec := range stream.Spec.Tags {
 		if tagSpec.Name == tag {
+			logrus.Printf("updating image ref for tag %s in stream %s with image %s", tag, stream.Name, imageRef)
```
Comment (bparees) [inline pkg/stub/jenkins.go]: should be debug level?

**Comment by bparees** [comment]
one nit, otherwise lgtm

File: pkg/stub/jenkins.go (hunk 1 above)
Comment (gabemontero) [reply pkg/stub/jenkins.go]: actually in the end I consciously chose non debug level ... I'd liked having this information in there

these messages are infrequent compared to the other messaging

still want me to make it debug?

Comment (bparees) [reply pkg/stub/jenkins.go]: falls under my statement from a few weeks ago that at some point we need to audit our default log messages, but for now it's ok.

**Comment by gabemontero** [comment]
ok e2e-aws-operator passed ... good news

**Pull request merged**